tests_integration/__tests__/doc-trim.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
+
+const docPrinter = require("../../src/doc/doc-printer");
+const docBuilders = require("../../src/doc/doc-builders");
+
+const printDocToString = docPrinter.printDocToString;
+const concat = docBuilders.concat;
+const line = docBuilders.line;
+const trim = docBuilders.trim;
+const group = docBuilders.group;
+const indent = docBuilders.indent;
+
+// These tests don't use `runPrettier` because `trim` is not used by any
+// bundled parser (only third-party plugins).
+
+describe("trim", () => {
+  test.each([
+    [
+      // trims the current line
+      group(concat(["hello    ", trim])),
+      "hello"
+    ],
+    [
+      // trims existing indentation
+      group(
+        concat([
+          "function()",
+          line,
+          "{",
+          indent(
+            concat([
+              line,
+              group(concat([trim, "#if DEBUG"])),
+              line,
+              "alert(42);",
+              line,
+              group(concat([trim, "#endif"]))
+            ])
+          ),
+          line,
+          "}"
+        ])
+      ),
+      `function()
+{
+#if DEBUG
+  alert(42);
+#endif
+}`
+    ],
+    [
+      // ignores trimmed characters when fitting the line
+      group(concat(["hello  ", "  ", trim, line, "world!"])),
+      "hello world!"
+    ]

```

Comment:
Could you add a test for trimming to the root (markAsRoot)?

Reply:
As is, the code will not respect `markAsRoot`, and will trim down to column 0. I confirmed this behavior with a test. I can commit that test, but I suppose your comment was about making the code respect `markAsRoot`.

Do you think it's worth dealing with this edge case? In which situation would I want to keep an indent of 4 spaces, but still want to trim the remaining spaces after?

Reply:
You should update the docs then, and also add a test to ensure this behavior. I guess we probably don't need to deal with it since there's probably no use case for it.

https://github.com/prettier/prettier/blob/33bceb0cfb3ea26180f132383bee574577e33437/commands.md#L240-L246